tools/transform_quic_trace.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A helper tool to perform various transformations on traces, notably
+// protobuf-to-JSON conversion and trace truncation.
+
+#include <iostream>
+#include <istream>
+
+#include "gflags/gflags.h"
+#include "google/protobuf/util/json_util.h"
+#include "lib/quic_trace.pb.h"
+
+DEFINE_bool(whitespace, true, "Add whitespace to the JSON output");
+DEFINE_string(input_format, "protobuf", "Can be 'protobuf' or 'json'.");
+DEFINE_string(output_format, "protobuf", "Can be 'protobuf' or 'json'.");
+DEFINE_int64(truncate_after_time_us,
+             -1,
+             "If positive, removes all events after the specified point in "
+             "time (us) from the trace.");
+
+namespace {
+
+using google::protobuf::util::JsonOptions;
+using google::protobuf::util::JsonStringToMessage;
+using google::protobuf::util::MessageToJsonString;
+
+bool IsValidFormatString(const std::string& format) {
+  return format == "protobuf" || format == "json";
+}
+
+bool InputTrace(quic_trace::Trace* trace) {
+  if (FLAGS_input_format == "json") {
+    std::istreambuf_iterator<char> it(std::cin);
+    std::istreambuf_iterator<char> end;
+
+    auto status = JsonStringToMessage(std::string(it, end), trace);
+    return status.ok();
+  }
+
+  return trace->ParseFromIstream(&std::cin);
+}
+
+void OutputTrace(const quic_trace::Trace& trace) {
```

Comment:
perhaps make this a bool, to percolate up whatever MessageToJsonString() returns

Reply:
I don't think it returns anything?